session.go

```diff
@@ -59,67 +113,84 @@ func fullProfile(app *App, user *User, uuid string, sign bool) (SessionProfileRe
 	}, nil
 }
 
-// /session/minecraft/hasJoined
-// https://c4k3.github.io/wiki.vg/Protocol_Encryption.html#Server
-func SessionHasJoined(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		playerName := c.QueryParam("username")
-		serverID := c.QueryParam("serverId")
+func (app *App) hasJoined(c *echo.Context, playerName string, serverID string, legacy bool) error {
+	var user User
+	result := app.DB.First(&user, "player_name = ?", playerName)
+	// If the error isn't "not found", throw.
+	if result.Error != nil && !errors.Is(result.Error, gorm.ErrRecordNotFound) {
+		return result.Error
+	}
 
-		var user User
-		result := app.DB.First(&user, "player_name = ?", playerName)
-		if result.Error != nil && !errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			if app.Config.TransientUsers.Allow && app.TransientUsernameRegex.MatchString(playerName) {
-				var err error
-				user, err = MakeTransientUser(app, playerName)
-				if err != nil {
-					return err
-				}
-			} else {
-				return c.NoContent(http.StatusForbidden)
+	if result.Error != nil || !user.ServerID.Valid || serverID != user.ServerID.String {
+		for _, fallbackAPIServer := range app.Config.FallbackAPIServers {
+			if fallbackAPIServer.DenyUnknownUsers && result.Error != nil {
+				// If DenyUnknownUsers is enabled and the player name is
+				// not known, don't query the fallback server.
+				continue
+			}
+			base, err := url.Parse(fallbackAPIServer.SessionURL)
```

Comment:
If we're going to with the simple step:
```go
			if legacy {
				// Replace string sessionserver.mojang.com with session.minecraft.net in the URL
				if strings.Contains(base.Host, "sessionserver.mojang.com") {
					base.Host = strings.ReplaceAll(base.Host, "sessionserver.mojang.com", "session.minecraft.net")
				}
			}
```

Reply:
It's simpler to use the newer route in all cases.

Reply:
I tried to use the newer route but seems like it refuses to even accept it, i just relied on legacy route again as it worked, unlike the modern route. Did it work for you?